csharp/src/Ice/Protocol.cs

```diff
@@ -21,14 +21,15 @@ public enum Protocol : byte
     internal static class ProtocolExtensions
     {
         /// <summary>Checks if this protocol is supported by the Ice runtime. If not supported, throws
-        /// UnsupportedProtocolException.</summary>
+        /// NotSupportedException.</summary>
         /// <param name="protocol">The protocol.</param>
         internal static void CheckSupported(this Protocol protocol)
         {
             // For now, we support only ice1
             if (protocol != Protocol.Ice1)
             {
-                throw new UnsupportedProtocolException("", protocol, Protocol.Ice1);
+                throw new System.NotSupportedException(
+                    $"Ice protocol `{protocol.ToString().ToLower()}' is not supported by this Ice runtime");
```

Comment:
Maybe we should include the runtime version here

```suggestion
                throw new System.NotSupportedException(
                    $"Ice protocol `{protocol.ToString().ToLower()}' is not supported by Ice `{Util.StringVersion()}' runtime");
```

Reply:
Fixed.